### PR TITLE
for v7-4-6 production

### PR DESCRIPTION
**Pull request by jshlee**
added both silver and golden weights
added flag in runData for silver and golden into pileweights
updated jet scale factors, JER and btagging
update electron eff area

### CODE REVIEW

**Comment by jhgoh** [comment]
Why don't you just have another producer? The lumi selection bit is not "pileup weight".

**Comment by jshlee** [comment]
i just put it as pileup weight for convenience, so the same variable can be used for both data and mc.

**Comment by jhgoh** [comment]
Maybe this would be less confusing with "lumiWeight".

File: CatProducer/plugins/PileupWeightProducer.cc [hunk 1]
```diff
@@ -106,9 +107,9 @@ PileupWeightProducer::PileupWeightProducer(const edm::ParameterSet& pset)
 void PileupWeightProducer::produce(edm::Event& event, const edm::EventSetup& eventSetup)
 {
   std::auto_ptr<int> nTrueIntr(new int(-1));
-  std::auto_ptr<float> weight(new float(1.));
-  std::auto_ptr<float> weightUp(new float(1.));
-  std::auto_ptr<float> weightDn(new float(1.));
+  std::auto_ptr<float> weight(new float(0.));
+  std::auto_ptr<float> weightUp(new float(0.));
+  std::auto_ptr<float> weightDn(new float(0.));
```
Comment (jhgoh) [inline CatProducer/plugins/PileupWeightProducer.cc]: Should be float(1.)

File: DataFormats/src/Jet.cc [hunk 2]
```diff
@@ -69,85 +72,106 @@ float Jet::scaleFactorCSVv2(Jet::BTAGCSV_CUT cutType, int syst, JETFLAV flav) co
...
+  if (cutType == 0 && syst == 0 && flav == 2 && pt > 20 && pt < 1000 ) return ((1.07278+(0.000535714*pt))+(-1.14886e-06*(pt*pt)))+(7.0636e-10*(pt*(pt*pt)));
+  if (cutType == 0 && syst == -1 && flav == 2 && pt > 20 && pt < 1000 ) return ((1.01637+(0.000265653*pt))+(-4.22531e-07*(pt*pt)))+(2.23396e-10*(pt*(pt*pt)));
+  if (cutType == 0 && syst == +1 && flav == 2 && pt > 20 && pt < 1000 ) return ((1.12921+(0.000804962*pt))+(-1.87332e-06*(pt*pt)))+(1.18864e-09*(pt*(pt*pt)));
+  if (cutType == 1 && syst == 0 && flav == 2 && pt > 20 && pt < 1000 ) return 1.14022;
+  if (cutType == 1 && syst == -1 && flav == 2 && pt > 20 && pt < 1000 ) return 0.94022;
+  if (cutType == 1 && syst == +1 && flav == 2 && pt > 20 && pt < 1000 ) return 1.34022;
+  if (cutType == 2 && syst == 0 && flav == 2 && pt > 20 && pt < 1000 ) return 0.907317;
+  if (cutType == 2 && syst == -1 && flav == 2 && pt > 20 && pt < 1000 ) return 0.557317;
+  if (cutType == 2 && syst == +1 && flav == 2 && pt > 20 && pt < 1000 ) return 1.257317;
```
Comment (jhgoh) [inline DataFormats/src/Jet.cc]: Should be pt >= 20 etc.
Do you think that this form is better than the previous one?

Comment (jshlee) [inline DataFormats/src/Jet.cc]: the previous one is better, but this was much easier to implement from their excel sheet.

**Pull request merged**